### ISSUE TITLE
Installing certificate error on Fedora (34)

### ISSUE THREAD

**Issue by emilbratt** [craftcms/nitro#380]
### Description

In the "trust part" of the nitro init command nitro is unable to install its root certificate in Fedora. After running the command "nitro init" everything works as intended all the way up until the part where it is about to install the root site certificate.

Here is my output from the terminal:
<pre>
..
...
Checking network…
  ✓ network ready
Checking proxy…
  ✓ proxy ready
Checking databases…
  … checking mysql-8.0-3306.database.nitro   … downloading mysql:8.0 ✓
Checking services…
  … checking dynamodb service ✓
  … checking mailhog service ✓
  … checking redis service ✓
Checking proxy…
  … updating proxy ✓
Updating hosts file (you might be prompted for your password)
[sudo] passord for bob: 
  … getting Nitro’s root site certificate ✓
Installing certificate (you might be prompted for your password)
Error: unable to find the distribution from the description: Description:	Fedora release 34 (Thirty Four)
</pre>
 
### Steps to reproduce

Run either of the below commands on Fedora 34
$ nitro init
$ nitro trust

### Additional info

- Nitro version: 2.0.8
- Distro: Fedora 34
The trust command that fails:
https://craftcms.com/docs/nitro/2.x/commands.html#trust

**Comment by jasonmccallister** [comment]
@emilbratt that logic is [stored here](https://github.com/craftcms/nitro/blob/develop/pkg/certinstall/certinstall_linux.go#L30). We default the finding `lsb_release` which is most likely not available on Fedora (I hopped back to PopOS so I cannot check).

Does your installation have a `cat /etc/issue` or `/etc/fedora-release`? If only there was a standard way to check on each distro :) 

Either way, we need to improve support for detecting a distribution.

**Comment by emilbratt** [comment]
> We default the finding lsb_release which is most likely not available on Fedora (I hopped back to PopOS so I cannot check).

That is correct. Fedora does not have /etc/lsb_release. Might be able to use this file instead /etc/os-release
https://www.freedesktop.org/software/systemd/man/os-release.html
<pre>
[bob@fedora]$ cat /etc/os-release 
NAME=Fedora
VERSION="34 (Workstation Edition)"
ID=fedora
VERSION_ID=34
VERSION_CODENAME=""
PLATFORM_ID="platform:f34"
PRETTY_NAME="Fedora 34 (Workstation Edition)"
...
...
</pre>

> Does your installation have a cat /etc/issue or /etc/fedora-release?

Yes, I have both files. This is the output:
<pre>
[bob@fedora]$ cat /etc/issue
\S
Kernel \r on an \m (\l)

[bob@fedora]$ cat /etc/fedora-release
Fedora release 34 (Thirty Four)
</pre>

> If only there was a standard way to check on each distro :)

Unfortunately I do not have any experience in the Go language.
However, I tend to do it this way when writing BASH scripts:
..cat out all file-names that contains the word "release" in /etc
cat /etc/*-release
..pipe it into a grep to search for "NAME"
grep -w "NAME"
..pipe into a cut where we strip the  "NAME" part of the match
cut -c 6-

The complete command:
<pre>
[bob@fedora]$ cat /etc/*-release | grep -w "NAME" | cut -c 6-
Fedora
</pre>
Or in BASH script
<pre>
distro=$(cat /etc/*-release | grep -w "NAME" | cut -c 6-)
</pre>

**Comment by jasonmccallister** [comment]
I'm also going to add Zorin OS to the list of supported distro's (which is just Ubuntu underneath).

**Comment by emilbratt** [comment]
> I'm also going to add Zorin OS to the list of supported distro's (which is just Ubuntu underneath).

That is great to hear!
May I also recommend adding Pop!_OS if possible? I have noticed some growing popularity of the distro. It s also Ubuntu underneath. :)

**Comment by jasonmccallister** [comment]
@emilbratt we just released 2.0.9 with improved support for Fedora, can you give it a try when you are free?

**Comment by emilbratt** [comment]
@jasonmccallister 
Sorry for the late feedback as I missed your notification
I tested this on Fedora just now.

### Testing New Version 2.0.9 on Fedora 35
After downloading tarball for x86_64 and verifying with sha256sum, i ran:
> $ nitro init
<pre>
..
...
Checking network…
  ✓ network ready
Checking proxy…
  ✓ proxy ready
Checking databases…
  … checking mysql-8.0-3306.database.nitro   … downloading mysql:8.0 ✓
  … checking postgres-13-5432.database.nitro   … downloading postgres:13 ✓
Checking services…
  … checking dynamodb ✓
  … checking mailhog ✓
  … checking redis ✓
Checking proxy…
  … updating proxy ✓
Updating hosts file (you might be prompted for your password)
[sudo] passord for bob: 
Error: exit status 1
</pre>
..this time the updating hosts part failed with exit 1

I also tried modifying the hosts with the dedicated command 
> sudo nitro hosts --hostnames myhostname
<pre>
Adding sites to hosts file…
  … modifying hosts file ✓
</pre>
..which worked as intended

I downloaded v. 2.0.8 because host-edit during init worked with previous version
To make sure verything was "clean" I removed v. 2.0.9 and all containers
> nitro destroy

Then running the 2.0.8 executable:
> nitro init

To my surprise the host-edit during nitro init did not work on 2.0.8 either.
The exact same output as above with the latest v. 2.0.9
May be related to Fedora 35 since this did not happen on Fedora 34


### The trust-part
Since the trust part comes after the host-edit part, I manually tested
this as well because the trust part was the original issue
> nitro trust
<pre>
[bob@fedora35 ~]$ nitro trust
  … getting Nitro’s root site certificate ✓
Installing certificate (you might be prompted for your password)
Nitro certificates are now trusted 🔒
</pre>
..and this seemed to be resolved


### Summary
- Nitro version: 2.0.9
- Distro: Fedora 35
- [x] #380 trusting root certificates now works
- [ ] host-edit during nitro init fails
- [x] host-edit using nitro hosts works

**Comment by jasonmccallister** [comment]
Hi, we are closing this issue as we have decided to retire Nitro, so no additional work will occur on this project. You can read the official blog post here https://craftcms.com/blog/retiring-craft-nitro. We appreciate everyones feedback and involvement and we look forward to refocusing our efforts on Cloud! 

If you're looking for a new local development environment, we recommend DDEV and have a knowledge base article to help you with the transition: https://craftcms.com/knowledge-base/migrating-from-craft-nitro-to-ddev.
